lib/start-server.js

```diff
@@ -37,23 +37,34 @@ const program = require('commander');
 
 program.option('--port <number>', 'Specify port number').parse(process.argv);
 
-const port = parseInt(program.port, 10) || 3000;
-
-tcpPortUsed
-  .check(port, 'localhost')
-  .then(function(inUse) {
-    if (inUse) {
-      console.error(chalk.red('Port ' + port + ' is in use'));
-      process.exit(1);
-    } else {
-      console.log('Starting Docusaurus server on port ' + port + '...');
-      // start local server on specified port
-      const server = require('./server/server.js');
-      server(port);
-    }
-  })
-  .catch(function(ex) {
-    setTimeout(function() {
-      throw ex;
-    }, 0);
-  });
+var port = process.env.PORT || 3000;
+var numAttempts = 0;
+var maxAttempts = 10;
+checkPort();
+
+function checkPort() {
+  tcpPortUsed
+    .check(port, 'localhost')
+    .then(function(inUse) {
+      if (inUse) {
+        console.error(chalk.red('Port ' + port + ' is in use'));
+        // Try again but with port + 1
+        port += 1;
+        numAttempts += 1;
+        checkPort();
+      } else if (inUse && numAttempts >= maxAttempts) {
```

Comment:
I don't think this will ever get executed since the first `inUse` check above will override it. I think you want the `numAttempts` check within the `if (inUse)` check above.

Reply:
Oops, this is embarrassing. One moment and I'll fix this.